File: packages/libapt-front/libapt-front-0.4.0/swig/ruby/apt-cat.rb
#!/usr/bin/ruby1.8

require "aptfront"

Aptfront::aptFrontInit()
cache = Aptfront::Global::cache()
puts cache
if (not cache.openCacheFile(Aptfront::OpProgress.new(), Aptfront::Cache::OpenReadOnly))
  puts "Error opening cache"
  error = Aptfront::getGlobalError()
  error.DumpErrors()
end

cache.openRecordsCache
pc = cache.packageCache
pl = pc.packageList

pl.each { |it|
  puts "Package: " + it.name
  #puts "Candidate Version: " + it.candidateVersion.versionString
  it.versionList.each { |v|
    puts "Version: " + v.versionString + (v == it.candidateVersion ? " (candidate)" : '') + (v == it.currentVersion ? " (current)" : '')
    puts "Section: " + v.section
    puts "Architecture: " + v.architecture
    puts "Description: " + v.shortDescription + v.longDescription
  }
  puts
}

File: packages/libapt-front/libapt-front-0.4.0/swig/python/apt-cat.py
#!/usr/bin/env python

import aptfront
import sys

aptfront.init()
cache = aptfront.getCache()
cache.open(aptfront.Cache.OpenDefault | aptfront.Cache.OpenReadOnly)

pc = cache.packages()

for pkg in pc:
    print "Package: " + pkg.name()
    for V in pkg:
        print "Version: " + V.versionString()
        print "Section: " + V.section()
        print "Architecture: " + V.architecture()
        print "Description: " + V.shortDescription() + V.longDescription()
